File: ZCopyUnitTester/PathFormatterTester.vb
Imports System.Text
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports ZCopy

<TestClass()>
Public Class PathFormatterTester

    <TestMethod()>
    Public Sub TestWithDoubleQoutesAndSlash()
        Dim input As String = "c:\user\joost\"""
        Dim expected As String = "c:\user\joost\"
        Assert.AreEqual(expected, PathFormatter.FormatPath(input))

    End Sub

    <TestMethod()>
    Public Sub TestWithDoubleQoutesNoSlash()
        Dim input As String = "c:\user\joost"""
        Dim expected As String = "c:\user\joost"
        Assert.AreEqual(expected, PathFormatter.FormatPath(input))

    End Sub

End Class

File: ZCopyUnitTester/PathFormatterTester.cs
using Xunit;
using ZCopy.Classes;

namespace ZCopyUnitTester
{
    [Collection("PathFormatterTester")]
    public class PathFormatterTester
    {
        [Fact()]
        public void TestWithDoubleQoutesAndSlash()
        {
            string input = @"c:\user\joost\""";
            string expected = @"c:\user\joost\";
            Assert.Equal(expected, PathFormatter.FormatPath(input));
        }

        [Fact()]
        public void TestWithDoubleQoutesNoSlash()
        {
            string input = @"c:\user\joost""";
            string expected = @"c:\user\joost";
            Assert.Equal(expected, PathFormatter.FormatPath(input));
        }
    }
}